### PR TITLE
feat-ManualInstallImplemented

### PR DESCRIPTION
**Pull request by prak24oct**

### CODE REVIEW

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt [hunk 1]
```diff
@@ -33,13 +34,26 @@ class TrackierSDKInstance {
...
+        }
+    }
+
+    fun setManualInstall() {
+        if (isManualInstall) {
+            scheduleInstall()
+        }
+    }
+
+    private fun scheduleInstall() {
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt:50]:
```suggestion
    private fun _fireInstall() {
```

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt [hunk 2]
```diff
@@ -33,13 +34,26 @@ class TrackierSDKInstance {
...
         this.appToken = this.config.appToken
         this.installId = getInstallID()
+        this.isManualInstall = config.getManualMode()
         DeviceInfo.init(device, this.config.context)
+        if (!isManualInstall) {
+            scheduleInstall()
+        }
+    }
+
+    fun setManualInstall() {
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt:44]:
```suggestion
    fun fireInstall() {
```

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt [hunk 3]
```diff
@@ -33,13 +34,26 @@ class TrackierSDKInstance {
         this.configLoaded = true
         this.appToken = this.config.appToken
         this.installId = getInstallID()
+        this.isManualInstall = config.getManualMode()
         DeviceInfo.init(device, this.config.context)
+        if (!isManualInstall) {
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt:39]:
```suggestion
        if (isManualInstall) {
return
```

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt [hunk 4]
```diff
@@ -33,13 +34,26 @@ class TrackierSDKInstance {
...
+            scheduleInstall()
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt:40]:
```suggestion
            _fireInstall()
```

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt [hunk 5]
```diff
@@ -33,13 +34,26 @@ class TrackierSDKInstance {
...
+        if (isManualInstall) {
+            scheduleInstall()
+        }
+    }
+
+    private fun scheduleInstall() {
         CoroutineScope(Dispatchers.IO).launch {
             initGaid()
             initAttributionInfo()
-            trackInstall()
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt:40]: TrackInstall pehle hona chahiye bhai fir hi session ki request jaayegi

**Review by Hemant-Mann — CHANGES_REQUESTED** [review]

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt (hunk 3 above)
Comment (prak24oct) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt:39]: In order to fire autoInstal for the case of not manual track , this has to be !isManualInstall as it is defined false initially

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt (hunk 5 above)
Comment (prak24oct) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt:40]: session has been now initialised in init method......

**Review by Hemant-Mann — APPROVED** [review]

**Pull request merged**